### PR TITLE
feat: add apple books highlight links

### PR DESCRIPTION
**Pull request by absorpheus**
**What is it?**

This feature will fetch the EPUB Canonical Fragment Identifier for highlights so that they can be opened at the exact highlight location with Apple Books.

**Summary**

* test(db): update migrations to enable new database tests

* test: add basic highlight links tests for db

* test: update mock tests, plugin docs and info

* refactor(core): update default template to include highlight location

* refactor(core): update types, constants, schemas, seedData and methods

* fix: template for colored highlights. register handlebars custom helpers. Minor grammar fixes

* fix: new lines no longer appear at the end of highlight blocks

* docs(README): update preview and template-colors screenshots to show highlight links feature

* docs(README): add highlight location template variable

* docs(README): update template with highlight link

* linter: update tsconfig to remove unnecessary type checks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,6 +133,7 @@ Number of annotations:: {{annotations.length}}
 {{#if (eq highlightStyle "4")}}- 🎯 Highlight:: <mark style="background:rgb(242,178,188); color:#000; padding:2px;">{{{highlight}}}</mark>{{/if}}
 {{#if (eq highlightStyle "5")}}- 🎯 Highlight:: <mark style="background:rgb(214,192,238); color:#000; padding:2px;">{{{highlight}}}</mark>{{/if}}
 - 📝 Note:: {{#if note}}{{{note}}}{{else}}N/A{{/if}}
+- 📙 Highlight Link:: {{#if highlightLocation}}[Apple Books Highlight Link](ibooks://assetid/{{../bookId}}#{{highlightLocation}}){{else}}N/A{{/if}}
```
Comment (bandantonio) [inline README.md:136]:
```suggestion
- 📙 Highlight Link:: {{#if highlightLocation}}[Apple Books Highlight Link](ibooks://assetid/{{bookId}}#{{highlightLocation}}){{else}}N/A{{/if}}
```

**Review by bandantonio — CHANGES_REQUESTED** [review]
@absorpheus Thank you for the contribution, the PR looks great.
However, I see multiple updates that you're trying to deliver:
- highlights link feature + tests and migrations
- fix for #25  (I will take care of that one, it needs tests to be fixed as well)
- fix for grammar 

I would recommend you to split this PR into two, each one containing the corresponding changes. Please use this repository directly (not a fork) to make those changes, because the required checks weren't triggered for some reason for the PR. I will investigate why.
Thank you.

File: template-colors.png
Comment (bandantonio) [inline template-colors.png:1]: I believe it's not necessary to add a highlight link to this template, as its main focus is on colors. So you can revert to the original screenshot.

**Comment by absorpheus** [comment]
@bandantonio Thank you for your feedback on the PR. I will work on making these changes very soon.

**Comment by bandantonio** [comment]
@absorpheus No worries, I just released a new version to fix the bug with custom templates, in order to prevent user experience disruption. So there is no rush, take your time, please.

**Comment by absorpheus** [comment]
@bandantonio Please give me permission to access the repository so I can push the feature branches and create the PR's. Thank you

**Comment by bandantonio** [comment]
This PR was split into several ones:
- highlights link feature + tests and migrations #27 
- fix for #25
- fix for grammar  #36

**Pull request closed without merging**